packages/cmrjs/cmr-utils.js

```diff
@@ -313,7 +278,19 @@ async function updateCMRMetadata(granuleId, cmrFile, files, distEndpoint, publis
   log.debug(`cmrjs.updateCMRMetadata granuleId ${granuleId}, cmrMetadata file ${cmrFile.filename}`);
 
   if (isECHO10File(cmrFile.filename)) {
-    return updateEcho10XMLMetadata(granuleId, cmrFile, files, distEndpoint, published);
+    const buckets = new BucketsConfig(await bucketsConfigDefaults());
+    const theMetadata = await updateEcho10XMLMetadata(cmrFile, files, distEndpoint, buckets);
+    if (published) {
+      // post metadata Object to CMR
+      const creds = getCreds();
+      const cmrFileObject = {
+        filename: cmrFile.filename,
+        metadataObject: theMetadata,
+        granuleId: granuleId
+      };
+      await publishECHO10XML2CMR(cmrFileObject, creds, process.env.bucket, process.env.stackName);
+    }
+    return Promise.resolve();
```

Comment:
So in the case of Echo10, it doesn't return the promise to upload updated CMR file (which it says it does in the docstring) and this is looking like echo10 and umm-G will have different return values.

Reply:
So this is interesting, I went back to the original code
https://github.com/nasa/cumulus/tree/eabf311c032a1d463e519f904d103e7816864569

The function `publishECHO10XML2CMR` was called `publish` and it was defined in `ingest/cmr.js` and called by two functions

`ingest/granule.js`: [link](https://github.com/nasa/cumulus/blob/eabf311c032a1d463e519f904d103e7816864569/packages/ingest/granule.js#L840)
```
  if (published) await publish(cmrFileObject, creds, process.env.bucket, process.env.stackName);
```

and `post-to-cmr/index.js`: [link](https://github.com/nasa/cumulus/blob/eabf311c032a1d463e519f904d103e7816864569/tasks/post-to-cmr/index.js#L70-L71)
```
// post all meta files to CMR
const publishRequests = cmrFiles.map((cmrFile) => publish(cmrFile, creds, bucket, stack));
const results = await Promise.all(publishRequests);

```

So It looks like even though the `publish` function returned an object
```
  return {
    granuleId: cmrFile.granuleId,
    filename: cmrFile.filename,
    conceptId,
    link: `${getUrl('search')}granules.json?concept_id=${res.result['concept-id']}`
  };
```

I think it was ignored by all of the code calling it. 🤷?  Should we do something different with the publish function or it's results?


As for the signatures differing between UMMG and ECHO10, that's probably a result of the refactoring to reuse the code in move-granules. I would expect them to look simliar by the time we finish.

Reply:
so clearly the doc-string is wrong...

Reply:
I think I could fix this to match the documentation by returning the value of publishECHO10XML2CMR, and updating the documentation, to say 
``` * @returns {Promise} returns promise to publish metadata to CMR Service or resolved promise if not published.```
with ```      return publishECHO10XML2CMR(cmrFileObject, creds, process.env.bucket, process.env.stackName);
```
Thoughts?

Reply:
what do you think of: 9cf3dfb5a?

Reply:
Yeah I think that's good. If the `published` flag is false and nothing is returned, will it break any calling functions?

Reply:
No, all of the existing calling functions are ignoring the return value, but they are all awaiting it so expect a promise.